app/models/alchemical_property.rb

```diff
@@ -1,9 +1,15 @@
 # frozen_string_literal: true
 
 class AlchemicalProperty < ApplicationRecord
-  has_many :alchemical_properties_canonical_ingredients, dependent: :destroy
-  has_many :canonical_ingredients, through: :alchemical_properties_canonical_ingredients
+  has_many :canonical_ingredients_alchemical_properties,
+           dependent:  :destroy,
+           class_name: 'Canonical::IngredientsAlchemicalProperty'
+  has_many :canonical_ingredients, through: :canonical_ingredients_alchemical_properties
 
   validates :name, presence: true, uniqueness: { message: 'must be unique' }
   validates :strength_unit, inclusion: { in: %w[point percentage], message: 'must be "point" or "percentage"', allow_blank: true }
+
+  def self.unique_identifier
```

Comment:
This method is used to determine whether items should be identified by name or item code (or some other attribute) in syncers.